### PR TITLE
Initial snmp_exporter setup

### PR DESCRIPTION
**Pull request by dkess**
This is a starting point for the snmp_exporter service. The main annoyance here is that the actual config file for the service must be "generated," see https://github.com/prometheus/snmp_exporter/tree/master/generator for details.

Thoughts on how I implemented this?

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+                stage('push-to-registry') {
+                    sh 'make push-image'
+                }
+
+                stage('deploy-to-prod') {
+                    build job: 'marathon-deploy-app', parameters: [
+                        [$class: 'StringParameterValue', name: 'app', value: 'snmp_exporter'],
+                        [$class: 'StringParameterValue', name: 'version', value: version],
+                    ]
```
Comment (kkuehlz) [inline Jenkinsfile]: we have code for [this](https://github.com/ocf/shared-pipeline/blob/ef3974ea4544cb6ce49fc31df8ef889f49ce42b6/vars/marathonDeployApp.groovy#L6)

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+} catch (err) {
+    def subject = "${env.JOB_NAME} - Build #${env.BUILD_NUMBER} - Failure!"
+    def message = "${env.JOB_NAME} (#${env.BUILD_NUMBER}) failed: ${env.BUILD_URL}"
+
+    if (env.BRANCH_NAME == 'master') {
+        slackSend color: '#FF0000', message: message
+        mail to: 'root@ocf.berkeley.edu', subject: subject, body: message
+    } else {
+        mail to: emailextrecipients([
```
Comment (kkuehlz) [inline Jenkinsfile]: and [this](https://github.com/ocf/shared-pipeline/blob/master/vars/emailNotification.groovy)

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	docker run --rm -p 9116:9116 "$(DOCKER_TAG)"
+
+gen-config: generator.yml vendor-snmp-exporter
+	docker run -ti \
+		-v $(PWD)/vendor-snmp-exporter/generator/mibs:/root/.snmp/mibs:ro \
+		-v $(PWD)/:/opt/ \
+		snmp-generator generate
+
+vendor-snmp-exporter:
+	git clone -q https://github.com/prometheus/snmp_exporter vendor-snmp-exporter
```
Comment (kkuehlz) [inline Makefile:24]: you are going to build off master instead of a specific tagged version?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,4 @@
+ARG snmp_exporter_version=latest
+FROM prom/snmp-exporter:${snmp_exporter_version}
```
Comment (kkuehlz) [inline Dockerfile:2]: if you are using this in one place and pulling the latest image is the variable really necessary?

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+	cd vendor-snmp-exporter/generator && make mibs
+	cd vendor-snmp-exporter/generator && docker build -t snmp-generator .
```
Comment (kkuehlz) [inline Makefile]:
```
cd vendor-snmp-exporter/generator && \ 
    make mibs && \
    docker build -t snmp-generator .
```

File: Dockerfile (hunk 4 above)
Comment (dkess) [reply Dockerfile:2]: This is standard in all of our Dockerfiles. The makefile overrides this with the version that we actually want.

File: Makefile (hunk 3 above)
Comment (dkess) [reply Makefile:24]: Will change this to a tag

File: Jenkinsfile (hunk 1 above)
Comment (dkess) [reply Jenkinsfile]: I'll switch to the declarative Jenkinsfile

**Review by abizer — APPROVED** [review]

**Pull request merged**